controller/src/lib.rs

```diff
@@ -21,12 +21,58 @@ pub mod run;
 mod scheduler;
 
 pub async fn run_scheduler(nats: TypedNats, state: StateHandle) -> NeverResult {
-    let scheduler = Scheduler::new(state);
+    let scheduler = Scheduler::new(state.clone());
     let mut schedule_request_sub = nats.subscribe(ScheduleRequest::subscribe_subject()).await?;
     tracing::info!("Subscribed to spawn requests.");
 
     while let Some(schedule_request) = schedule_request_sub.next().await {
         tracing::info!(spawn_request=?schedule_request.value, "Got spawn request");
+
+        if let Some(lock) = &schedule_request.value.lock {
```

Comment:
I think it'd be easier to read if this were factored into a handle_locked_request or something.